sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -359,4 +375,31 @@ trait GpuHashJoin extends GpuExec {
       joinedTable.close()
     }
   }
+
+  // Filter null values for build-side keys, so as to ensure no nullable join column included in
+  // built table if compareNullsEqual is true.
+  protected def filterBuiltNullsIfNecessary(table: Table): Table = closeOnExcept(table) { t =>
+    if (compareNullsEqual && gpuBuildKeys.exists(_.nullable)) {
+      filterNulls(t, joinKeyIndices, closeTable = true)
+    } else {
+      t
+    }
+  }
+
+  private[this] def anyNullableStructChild(expressions: Seq[GpuExpression]): Boolean = {
```

Comment:
nit: An explanation of why we need this might be helpful here. Right now we know that joins have issues with struct children, but it would be good to document what the issue is and if there are any follow on issues to address it.

Reply:
Comments appended.